### DISCUSSION TITLE
Dynamic loading of rule from DB directly and update on daily basis for new rule add/delete in DB.

### DISCUSSION THREAD

**Discussion by mannu717** [evrete/evrete#74] [Q&A]
Hi Team, 
I have a requirement where I need to load business rules from the Database and execute them, and daily I want to update my rule engine with newly added/removed rules in DB, how can I do that?

Also if you have any example of How to create a custom DSL for Evrete, please share.

I'd appreciate any advice!

**Comment by andbi** [comment]
Hi, the library's `Knowledge` is not serializable , one can not save it directly in a database. All you need is serializable DSL (domain-specific language) which will be stored in the database and converted to the engine's `Knowledge` on retrieval. This cab be an XML, Excel, or any other format you choose for your rules.

Please have a look at this thread on  the general approach and why Evrete does not provide such storage formats out of the box.
[Issue 73](https://github.com/evrete/evrete/issues/73)
